library/std/src/sys/sgx/abi/mem.rs

```diff
@@ -21,8 +21,15 @@ extern "C" {
 #[inline(always)]
 #[unstable(feature = "sgx_platform", issue = "56975")]
 pub fn image_base() -> u64 {
-    let base;
-    unsafe { llvm_asm!("lea IMAGE_BASE(%rip),$0":"=r"(base)) };
+    let base: u64;
+    unsafe {
+        asm!(
+            "lea IMAGE_BASE(%rip), {}",
+            lateout(reg) base,
+            // FIXME(#76738): ATT syntax is used to support LLVM 8 and 9.
+            options(att_syntax, nostack),
```

Comment:
Also: `preserves_flags`, `nomem`.

Does `nomem` imply `nostack` and `readonly`?

Reply:
`nostack` is separate from `nomem`: you can still allocate and use stack memory with `nomem` as long as you don't access any memory outside the `asm`.